Add RadioUnit executor and receiver boilerplate

matthewkwsiu opened the pull request:
Adds basic methods and interfaces for the RadioUnit executor
and receiver boilerplate code.

The current code does not implement functionality, but has all the
methods as described in the UML diagram.

Main goals for reviewing are:
- David: double-check that we're implementing class relationships correctly
- Anyone: Checking for typos

Commit:
Add RadioUnit executor and receiver boilerplate

Added basic methods and interfaces for the RadioUnit executor
and receiver boilerplate code.

The current code does not implement functionality, but has all the
methods as described in the UML diagram.

Commit:
Add RadioUnit executor and receiver boilerplate

Added basic methods and interfaces for the RadioUnit executor
and receiver boilerplate code.

The current code does not implement functionality, but has all the
methods as described in the UML diagram.

Commit:
Merge branch 'receiverAndExecutorBoilerplate' of https://github.com/RPahlevan/OODesignFinal2021 into receiverAndExecutorBoilerplate

dLeung09 commented on src/main/java/radiounit_executorandreceiver/EricssonLteCommandExecutor.java:
```diff
@@ -0,0 +1,75 @@
+package radiounit_executorandreceiver;
```
Minor: I 'd say let's just leave this in the `radiounit` package - might be confusing if we add too many packages

matthewkwsiu replied:
Sounds good. I'll move them.

dLeung09 commented on src/main/java/radiounit_executorandreceiver/RadioCommandExecutorFactory.java:
```diff
@@ -0,0 +1,12 @@
+ * @author esiumat
+ *
+ */
+public abstract class RadioCommandExecutorFactory {
```
I can't recall if there was a specific reason we're using an abstract class here (instead of an interface), but based on the fact that we're just exposing a single public method, we could probably get away with an interface - we can implement many interfaces, but only inherit from one abstract class

matthewkwsiu replied:
Yeah, I think this is true. I defaulted to abstract since it was the class used in the refactoring guru factory UML. 

I see the benefit of your suggestion. I'll change it.

matthewkwsiu replied:
I suppose similarly our RadioUnitReceiverFactory also only exposes one public method. I'll adjust that to an interface as well.

dLeung09 commented on src/main/java/radiounit_executorandreceiver/NokiaWcdmaComandExecutorFactory.java:
```diff
@@ -0,0 +1,17 @@
+ * @author esiumat
+ *
+ */
+public class NokiaWcdmaComandExecutorFactory extends RadioCommandExecutorFactory {
```
typo: `NokiaWcdmaComandExecutorFactory` -> `NokiaWcdmaCommandExecutorFactory` (missing an 'm' in Command)

matthewkwsiu replied:
Gotcha, thanks for catching that

dLeung09 reviewed (approved):
Looks good - a few minor comments, but nothing that can't be submitted

Commit:
Adjust abstract to interface and update package

Changed abstract classes RadioCommandExecutorFactory and
RadioUnitReceiverFactory to interfaces. Moved classes to
RadioUnit package.
